rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +95,262 @@ typedef struct
+    Label() { _label_ids = {-1}; }
+    std::vector<int>& get_labels() { return _label_ids; }
+    void set_labels(Labels label_ids) override { _label_ids = std::move(label_ids); }
+    BoundingBoxCords& get_bb_cords() override { THROW("Not Implemented") }
+    void set_bb_cords(BoundingBoxCords bb_cords) override { THROW("Not Implemented") }
+    std::vector<int>& get_polygon_count() override { THROW("Not Implemented") }
+    std::vector<std::vector<int>>& get_vertices_count() override { THROW("Not Implemented") }
+    MaskCords& get_mask_cords() override { THROW("Not Implemented") }
+    void set_mask_cords(MaskCords mask_cords) override { THROW("Not Implemented") }
+    void set_polygon_counts(std::vector<int> polygon_count) override { THROW("Not Implemented") }
+    void set_vertices_counts(std::vector<std::vector<int>> vertices_count) override { THROW("Not Implemented") }
+    JointsData& get_joints_data() override { THROW("Not Implemented") }
+    void set_joints_data(JointsData *joints_data) override { THROW("Not Implemented") }
+protected:
+    Labels _label_ids = {}; // For label use only
 };
 
-struct BoundingBox : public MetaData
+class BoundingBox : public Label
 {
+public:
     BoundingBox()= default;
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids)
+    BoundingBox(BoundingBoxCords bb_cords, Labels bb_label_ids, ImgSize img_size = ImgSize{0, 0}, uint img_id = 0)
     {
         _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
+        _label_ids = std::move(bb_label_ids);
+        _info.img_size = std::move(img_size);
+        _info.img_id = img_id;
     }
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids, ImgSize img_size)
-    {
-        _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
-        _img_size = std::move(img_size);
-    }
-    void set_bb_cords(BoundingBoxCords bb_cords) { _bb_cords =std::move(bb_cords); }
-    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, BoundingBoxLabels bb_label_ids)
-    {
-        _bb_cords_xcycwh =std::move(bb_cords_xcycwh);
-        _bb_label_ids = std::move(bb_label_ids);
-    }
-    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, BoundingBoxLabels bb_label_ids, ImgSize img_size)
+    BoundingBoxCords& get_bb_cords() override { return _bb_cords; }
+    void set_bb_cords(BoundingBoxCords bb_cords) override { _bb_cords = std::move(bb_cords); }
+protected:
+    BoundingBoxCords _bb_cords = {}; // For bb use
+};
+
+struct PolygonMask : public BoundingBox {
+public:
+    PolygonMask(BoundingBoxCords bb_cords, Labels bb_label_ids, ImgSize img_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count)
     {
-        _bb_cords_xcycwh =std::move(bb_cords_xcycwh);
-        _bb_label_ids = std::move(bb_label_ids);
-        _img_size = std::move(img_size);
+        _bb_cords = std::move(bb_cords);
```

Comment:
be aware that this will copy all the meta_data vectors into the class members. Is this what is intended